plugins/feedback/src/components/EntityFeedbackPage/EntityFeedbackPage.tsx

```diff
@@ -0,0 +1,192 @@
+import React, { useEffect, useState } from 'react';
+
+import { Progress } from '@backstage/core-components';
+import { identityApiRef, useApi } from '@backstage/core-plugin-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+
+import {
+  ButtonGroup,
+  CircularProgress,
+  createStyles,
+  Dialog,
+  makeStyles,
+  Snackbar,
+  Theme,
+  Tooltip,
+  Zoom,
+} from '@material-ui/core';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import Add from '@material-ui/icons/Add';
+import ArrowForwardRounded from '@material-ui/icons/ArrowForwardRounded';
+import Sync from '@material-ui/icons/Sync';
+import { Alert } from '@material-ui/lab';
+
+import { CreateFeedbackModal } from '../CreateFeedbackModal/CreateFeedbackModal';
+import { FeedbackDetailsModal } from '../FeedbackDetailsModal';
+import { FeedbackTable } from '../FeedbackTable';
+import { CustomEmptyState } from './CustomEmptyState';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    buttonGroup: {
+      textAlign: 'center',
+      whiteSpace: 'nowrap',
+      marginTop: theme.spacing(1),
+    },
+  }),
+);
+
+export const EntityFeedbackPage = () => {
+  const classes = useStyles();
+  const { entity } = useEntity();
+  const user = useApi(identityApiRef);
+
+  const [modalProps, setModalProps] = useState<{
+    projectEntity: string;
+    userEntity: string;
+    serverType: string;
+  }>();
+
+  const [modalOpen, setModalOpen] = useState(false);
+  const [snackbarOpen, setSnackbarOpen] = useState<{
+    message: string;
+    open: boolean;
+    severity: 'success' | 'error';
+  }>({
+    message: '',
+    open: false,
+    severity: 'success',
+  });
+  const [reload, setReload] = useState(false);
+  useEffect(() => {
+    setReload(false);
+  }, [snackbarOpen, reload]);
+
+  const projectEntity =
+    `${entity.kind}:${entity.metadata.namespace}/${entity.metadata.name}`.toLowerCase();
+
+  const pluginConfig = {
+    'feedback/type': entity.metadata.annotations!['feedback/type'],
+    'feedback/host': entity.metadata.annotations!['feedback/host'],
+    'feedback/mail-to': entity.metadata.annotations!['feedback/email-to'],
+    'jira/project-key': entity.metadata.annotations!['jira/project-key'],
+  };
+
+  async function handleModalOpen() {
+    const userEntity = (await user.getBackstageIdentity()).userEntityRef;
+    setModalProps({
+      projectEntity: projectEntity,
+      userEntity: userEntity,
+      serverType: pluginConfig['feedback/type'],
+    });
+
+    return setModalOpen(true);
+  }
+
+  const handleModalClose = (respObj?: any) => {
```

Comment:
it'd be better to use a proper type here, as noted below if `false` is passed to this function instead of an empty string it might make the intent of the function a little bit clearer, but annotating this type properly would help your future self when a bug needs to be fixed in this area.

Reply:
Thanks for pointing it out, I have added type for `respObj` and also the case for `false`